### PR TITLE
Add updates to tests/e2e

### PR DESCRIPTION
**Pull request by justinsb**
This ensures that we get a bit more test coverage here.

### CODE REVIEW

File: config/tests/samples/create/samples.go [hunk 1]
```diff
@@ -90,17 +90,44 @@ func getNamespaces(samples []Sample) []string {
...
+	waitForReady(t, opt.Create)
+
+	if len(opt.Updates) != 0 {
+		// treat as a patch
+		for _, updateUnstruct := range opt.Updates {
+			if err := t.GetClient().Patch(ctx, updateUnstruct, client.Apply, client.FieldOwner("kcc-tests"), client.ForceOwnership); err != nil {
+				t.Fatalf("error updating resource: %v", err)
+			}
+		}
+		time.Sleep(5 * time.Second)
```
Comment (cheftako) [inline config/tests/samples/create/samples.go:123]: :( This seems like a flake waiting to happen. Can we switch to something which stops waiting early when success is reached but has a longer timeout for failure? I know K8s has utilities like that. Not saying you have to use those utilities but ....

Comment (justinsb) [reply config/tests/samples/create/samples.go:123]: Fair - it's not ideal.  We need status.observedGeneration to do this "right" (that's why we're waiting here).  I'll check out if we have that field today....

Comment (justinsb) [reply config/tests/samples/create/samples.go:123]: Good news, we do have observedGeneration and it looks easy to improve our wait-for-ready logic to incorporate it: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/871

Comment (justinsb) [reply config/tests/samples/create/samples.go:123]: I've removed this sleep.  It's not 100% safe until we get #871 in, but it should "fail open" I believe

**Comment by justinsb** [comment]
@diviner524 would be great to get this one in, without it I'm reluctant to mark other PRs that extend mockgcp in (because we aren't testing the update method without this PR)

File: config/tests/samples/create/samples.go [hunk 2]
```diff
@@ -90,17 +90,42 @@ func getNamespaces(samples []Sample) []string {
 	return results
 }
 
-func RunCreateDeleteTest(t *Harness, unstructs []*unstructured.Unstructured, cleanupResources bool) {
+type CreateDeleteTestOptions struct {
+	// Create is the set of objects to create
+	Create []*unstructured.Unstructured
+
+	// Updates is the set of objects to update (after all objects have been created)
+	Updates []*unstructured.Unstructured
```
Comment (diviner524) [inline config/tests/samples/create/samples.go:98]: * Where do we plan to store these updates (assuming they are going to be another set of YAML files with slight changes)? If they are going to be under `samples/resources/` we need to make sure they don't get rendered in the resource reference page.

* Also we are already testing updates using artifacts under `pkg/test/resourcefixture/testdata/basic`.
For example: https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/pkg/test/resourcefixture/testdata/basic/certificatemanager/v1beta1/certificatemanagercertificate/certificatemanagercertificatemanageddns

Would it be easier to directly enable mock for these test cases?

Comment (justinsb) [reply config/tests/samples/create/samples.go:98]: Yes, this is leveraging the existing fixture/sample data.  Where those have updates, we will now do those updates.

I don't mind us trying to enable mocks with the "old" tests, they're just a little more complicated to deal with, so I found it easier to just build up a new set of tests for the mock tests.

Comment (diviner524) [reply config/tests/samples/create/samples.go:98]: > Where those have updates, we will now do those updates.

Are you referring to data under `config/samples/resources` or other test data? I don't think we have any updates under `config/samples/resources`.

Comment (justinsb) [reply config/tests/samples/create/samples.go:98]: You're right, looks like we only set Update on the fixtures test data, not the samples test data.

**Comment by justinsb** [comment]
Rebased

**Comment by justinsb** [comment]
cc @diviner524 this one is holding up the rest of the parade, because without it we don't require the update method (so it has to keep growing as we add more resources)

**Review by gemmahou — APPROVED** [review]

**Pull request merged**